### PR TITLE
fix(mural): send API-compliant arrow widget payload

### LINKED ISSUE #1941: fix: Mural CLI arrow create sends payload shape rejected by API
### Component

Skills

## Summary

Arrow widget creation sends endpoint fields (`x1`, `y1`, `x2`, `y2`) that are rejected by the Mural API endpoint. The API expects origin and dimensions with relative points.

## Observed Behavior

* Command fails with HTTP 400 invalid payload
* Failure occurs when creating arrows via CLI even with valid coordinates
* Example command:

```bash
python -m mural widget create arrow --mural <mural-id> --x1 350 --y1 510 --x2 700 --y2 510
```

## Impact

* Arrow creation is blocked in automation flows
* Limits ability to programmatically connect widgets and annotate flows
* Causes confusion because coordinate inputs appear valid

## Suggested Fix

Normalize CLI coordinate inputs to Mural API arrow schema:

1. Parse `x1`, `y1`, `x2`, `y2` as numeric endpoints
2. Compute origin:
   * `x = min(x1, x2)`
   * `y = min(y1, y2)`
3. Compute dimensions with non-zero lower bound:
   * `width = max(abs(x2 - x1), 1.0)`
   * `height = max(abs(y2 - y1), 1.0)`
4. Compute relative points:
   * `[{"x": x1 - x, "y": y1 - y}, {"x": x2 - x, "y": y2 - y}]`
5. Keep `style` passthrough behavior unchanged

## Affected Area

* `.github/skills/experimental/mural/scripts/mural/_validation.py`
* `.github/skills/experimental/mural/tests/test_mural_helpers.py`
* `.github/skills/experimental/mural/tests/test_mural_commands.py`

### Expected Behavior

* CLI should serialize arrows in API-compliant shape
* Arrow creation should succeed for valid coordinates
* Optional style payload should remain supported

## Environment

* OS: Windows
* Shell: PowerShell
* Python environment: `C:\GitHub\hve-core\.venv`
* Module path: `.github/skills/experimental/mural/scripts`

## Steps To Reproduce

1. Authenticate successfully with mural CLI
2. Use any accessible mural ID
3. Run:

```bash
python -m mural widget create arrow --mural <mural-id> --x1 350 --y1 510 --x2 700 --y2 510
```

4. Observe request failure

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by akazlow**
# Pull Request

## Description
Fixes the Mural arrow widget payload shape to match API expectations.

The payload builder now emits normalized origin coordinates, non-zero bounding box dimensions, and points anchored to the normalized origin so arrow creation requests pass schema validation.

## Related Issue(s)
Fixes #1941

## Type of Change

Select all that apply:

**Code & Documentation:**

* [x] Bug fix (non-breaking change fixing an issue)
* [ ] New feature (non-breaking change adding functionality)
* [ ] Breaking change (fix or feature causing existing functionality to change)
* [ ] Documentation update

**Infrastructure & Configuration:**

* [ ] GitHub Actions workflow
* [ ] Linting configuration (markdown, PowerShell, etc.)
* [ ] Security configuration
* [ ] DevContainer configuration
* [ ] Dependency update

**AI Artifacts:**

* [ ] Reviewed contribution with `prompt-builder` agent and addressed all feedback
* [ ] Copilot instructions (`.github/instructions/*.instructions.md`)
* [ ] Copilot prompt (`.github/prompts/*.prompt.md`)
* [ ] Copilot agent (`.github/agents/*.agent.md`)
* [ ] Copilot skill (`.github/skills/*/SKILL.md`)

**Other:**

* [x] Script/automation (`.ps1`, `.sh`, `.py`)
* [ ] Other (please describe):

## Testing
- Updated and expanded tests in `.github/skills/experimental/mural/tests/test_mural_helpers.py` for:
  - reversed X direction normalization (`x1 > x2`)
  - vertical arrow clamping (`x1 == x2`)
  - horizontal arrow clamping (`y1 == y2`)
- Updated command-level tests in `.github/skills/experimental/mural/tests/test_mural_commands.py` for payload contract alignment.
- Local pytest execution is currently blocked in this shell due to missing `pytest` in the active `.venv`; CI remains the source of truth for full test execution.

### Required Checks

* [x] Documentation is updated (if applicable)
* [x] Files follow existing naming conventions
* [x] Changes are backwards compatible (if applicable)
* [x] Tests added for new functionality (if applicable)

### AI Artifact Contributions
* [ ] Used `/prompt-analyze` to review contribution
* [ ] Addressed all feedback from `prompt-builder` review
* [ ] Verified contribution follows common standards and type-specific requirements

### Required Automated Checks

The following validation commands must pass before merging:

* [x] Markdown linting: `npm run lint:md`
* [x] Spell checking: `npm run spell-check`
* [ ] Frontmatter validation: `npm run lint:frontmatter`
* [x] Skill structure validation: `npm run validate:skills`
* [ ] Link validation: `npm run lint:md-links`
* [ ] PowerShell analysis: `npm run lint:ps`
* [ ] Plugin freshness: `npm run plugin:generate`
* [ ] Docusaurus tests: `npm run docs:test`

## Security Considerations
* [x] This PR does not contain any sensitive or NDA information
* [x] Any new dependencies have been reviewed for security issues
* [x] Security-related scripts follow the principle of least privilege

## Additional Notes
- Addressed review feedback by adding a clarifying clamp-intent code comment in `_build_arrow_body` and adding edge-case test coverage for degenerate dimensions.

### CODE REVIEW

**Comment by akazlow** [comment]
## Review Summary

This fix corrects arrow widget payload schema to match Mural API requirements.

### Key Changes
- **_validation.py**: Updated arrow payload builder to emit required geometry fields
- **test_mural_commands.py** + **test_mural_helpers.py**: Added/updated test coverage for arrow payload shape

### Impact
- Arrow widget creation requests now pass Mural API validation
- Prevents silent failures due to malformed payloads
- Test suite validates both old and new payload contracts

### Verification Checklist
- [x] Geometry fields match Mural API schema
- [x] Test coverage spans both builder and helpers
- [x] Payload shape validated in command flow
- [x] No regression on other widget types

File: .github/skills/experimental/mural/scripts/mural/_validation.py [hunk 1]
```diff
@@ -475,11 +475,23 @@ def _build_shape_body(args: argparse.Namespace) -> dict[str, Any]:
...
 
 def _build_arrow_body(args: argparse.Namespace) -> dict[str, Any]:
+    x1 = _coerce_xy(getattr(args, "x1", None), "--x1")
+    y1 = _coerce_xy(getattr(args, "y1", None), "--y1")
+    x2 = _coerce_xy(getattr(args, "x2", None), "--x2")
+    y2 = _coerce_xy(getattr(args, "y2", None), "--y2")
+    origin_x = min(x1, x2)
+    origin_y = min(y1, y2)
+    width = max(abs(x2 - x1), 1.0)
+    height = max(abs(y2 - y1), 1.0)
```
Comment (rezatnoMsirhC) [inline .github/skills/experimental/mural/scripts/mural/_validation.py:487]: Nit: when `x1 == x2`, `width` is clamped to `1.0` to prevent zero-dimension bounding boxes, but both `points[0].x` and `points[1].x` remain `0.0`. The declared width and the point span are inconsistent in this degenerate case. A brief comment here would help future maintainers understand the intent, for example:

```python
# width/height are clamped to 1.0 to prevent zero-dimension bounding boxes
# that the Mural API may reject; point coordinates remain accurate.
width = max(abs(x2 - x1), 1.0)
height = max(abs(y2 - y1), 1.0)
```

File: .github/skills/experimental/mural/tests/test_mural_helpers.py
Comment (rezatnoMsirhC) [inline .github/skills/experimental/mural/tests/test_mural_helpers.py:1]: Nit: the existing test covers only the happy path with non-zero, non-reversed coordinates. A few additional cases would strengthen confidence in the builder:

- `x1 > x2` (reversed horizontal direction, verifies origin normalisation)
- `x1 == x2` (vertical arrow, `width` clamped to `1.0`)
- `y1 == y2` (horizontal arrow, `height` clamped to `1.0`)

These are especially useful given the clamping logic in `_build_arrow_body`.

**Review by rezatnoMsirhC — COMMENTED** [review]
Thank you for your contribution! Before we proceed with the review, could you please update the PR description to follow the [repository's PR template](https://github.com/microsoft/hve-core/blob/main/.github/PULL_REQUEST_TEMPLATE.md)?

A few things to address:

- Replace the current headings with the template's sections: `## Description`, `## Related Issue(s)`, `## Type of Change`, `## Testing`, `## Checklist`, and `## Security Considerations`.
- Check the appropriate boxes under **Type of Change** (this looks like a "Bug fix" and "Script/automation" change).
- Move `Fixes #1941` into the `## Related Issue(s)` section.
- Fill in the `## Testing` section describing how you tested the changes.
- Complete the `## Checklist` and `## Security Considerations` checkboxes.

No rush, and thanks again for taking the time to fix this!

**Comment by akazlow** [comment]
> Thank you for your contribution! Before we proceed with the review, could you please update the PR description to follow the [repository's PR template](https://github.com/microsoft/hve-core/blob/main/.github/PULL_REQUEST_TEMPLATE.md)?
> 
> A few things to address:
> 
> * Replace the current headings with the template's sections: `## Description`, `## Related Issue(s)`, `## Type of Change`, `## Testing`, `## Checklist`, and `## Security Considerations`.
> * Check the appropriate boxes under **Type of Change** (this looks like a "Bug fix" and "Script/automation" change).
> * Move `Fixes #1941` into the `## Related Issue(s)` section.
> * Fill in the `## Testing` section describing how you tested the changes.
> * Complete the `## Checklist` and `## Security Considerations` checkboxes.
> 
> No rush, and thanks again for taking the time to fix this!

Can do!

File: .github/skills/experimental/mural/scripts/mural/_validation.py (hunk 1 above)
Comment (akazlow) [reply .github/skills/experimental/mural/scripts/mural/_validation.py:487]: Thanks! Great feedback. Will update

Comment (akazlow) [reply .github/skills/experimental/mural/scripts/mural/_validation.py:487]: addressed

File: .github/skills/experimental/mural/tests/test_mural_helpers.py
Comment (akazlow) [reply .github/skills/experimental/mural/tests/test_mural_helpers.py:1]: addressed

**Review by rezatnoMsirhC — APPROVED** [review]

**Review by WilliamBerryiii — APPROVED** [review]
